File: Design and Programming Environments/Iteration 2/TicketManagerCSharp/TicketManagerCSharp/domain/Ticket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TicketManagerCSharp.domain
{
    class Ticket:HasId<int>
    {
        private int id;
        private int idEvent;
        private string buyer;
        private double price;

        public Ticket(int id, int idEvent, string buyer, double price)
        {
            this.id = id;
            this.idEvent = idEvent;
            this.buyer = buyer;
            this.price = price;
        }

        public Ticket(int idEvent, string buyer, double price)
        {
            this.idEvent = idEvent;
            this.buyer = buyer;
            this.price = price;
        }

        public int Id
        {
            get { return id; }
            set { id = value; }
        }

        public int IdEvent
        {
            get { return idEvent; }
            set { idEvent = value; }
        }

        public string Buyer
        {
            get { return buyer; }
            set { buyer = value; }
        }

        public double Price
        {
            get { return price; }
            set { price = value; }
        }

        public override bool Equals(object obj)
        {
            var ticket = obj as Ticket;
            return ticket != null &&
                   id == ticket.id &&
                   idEvent == ticket.idEvent &&
                   buyer == ticket.buyer &&
                   price == ticket.price;
        }

        public override int GetHashCode()
        {
            var hashCode = 791292794;
            hashCode = hashCode * -1521134295 + id.GetHashCode();
            hashCode = hashCode * -1521134295 + idEvent.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(buyer);
            hashCode = hashCode * -1521134295 + price.GetHashCode();
            return hashCode;
        }

        public override string ToString()
        {
            return "Ticket{" +
                "id=" + id +
                ", event=" + idEvent +
                ", buyer='" + buyer + '\'' +
                ", price=" + price +
                '}';
        }
    }
}

File: Design and Programming Environments/Iteration 3/TicketManager/src/main/java/Domain/Ticket.java
package Domain;

import java.util.Objects;

public class Ticket {
    private Integer id;
    private Integer event;
    private String buyer;
    private Double price;

    public Ticket(Integer id, Integer event, String buyer, Double price) {
        this.id = id;
        this.event = event;
        this.buyer = buyer;
        this.price = price;
    }

    public Ticket(Integer event, String buyer, Double price) {
        this.event = event;
        this.buyer = buyer;
        this.price = price;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getEvent() {
        return event;
    }

    public void setEvent(Integer event) {
        this.event = event;
    }

    public String getBuyer() {
        return buyer;
    }

    public void setBuyer(String buyer) {
        this.buyer = buyer;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Ticket ticket = (Ticket) o;
        return Objects.equals(id, ticket.id) &&
                Objects.equals(event, ticket.event) &&
                Objects.equals(buyer, ticket.buyer) &&
                Objects.equals(price, ticket.price);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "id=" + id +
                ", event=" + event +
                ", buyer='" + buyer + '\'' +
                ", price=" + price +
                '}';
    }
}
